overlord/snapshotstate/snapshotstate.go

```diff
@@ -330,6 +343,13 @@ func Import(ctx context.Context, st *state.State, r io.Reader) (setID uint64, sn
 		if dupErr, ok := err.(backend.DuplicatedSnapshotImportError); ok {
 			st.Lock()
 			defer st.Unlock()
+			// XXX: ideally, we should retry import.
+			if err := checkSnapshotConflict(st, dupErr.SetID, "forget-snapshot"); err != nil {
+				return 0, nil, err
+			}
+			opDone := setSnapshotOpInProgress(st, dupErr.SetID, "import-snapshot")
```

Comment:
do we need to do this, given that we are holding the state lock?

Reply:
We don't need this; I thought it would make sense for symmetry with export. Added a comment but I can drop it completely if you think it doesn't make sense (in which case I'd just add a comment explaining why it's not needed, to avoid confusion in the future).